lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -0,0 +1,157 @@
+"""
+This file contains (or should), all entrance exam related utils/logic.
+"""
+from django.conf import settings
+
+from courseware.access import has_access
+from courseware.model_data import FieldDataCache
+from courseware.models import StudentModule
+from opaque_keys.edx.keys import UsageKey
+from student.models import EntranceExamConfiguration
+from util.milestones_helpers import get_required_content
+from util.module_utils import yield_dynamic_descriptor_descendents
+from xmodule.modulestore.django import modulestore
+
+
+def feature_is_enabled():
+    """
+    Checks to see if the Entrance Exams feature is enabled
+    Use this operation instead of checking the feature flag all over the place
+    """
+    return settings.FEATURES.get('ENTRANCE_EXAMS', False)
+
+
+def course_has_entrance_exam(course):
+    """
+    Checks to see if a course is properly configured for an entrance exam
+    """
+    if not feature_is_enabled():
+        return False
+    if not course.entrance_exam_enabled:
+        return False
+    if not course.entrance_exam_id:
+        return False
+    return True
+
+
+def user_can_skip_entrance_exam(request, user, course):
+    """
+    Checks all of the various override conditions for a user to skip an entrance exam
+    """
+    if not course_has_entrance_exam(course):
+        return True
+    if user.is_anonymous():
+        return False
+    if has_access(user, 'staff', course):
+        return True
+    if EntranceExamConfiguration.user_can_skip_entrance_exam(user, course.id):
+        return True
+    exam_content = get_entrance_exam_content(request, course)
+    if not exam_content:
+        return True
+    return False
+
+
+def user_has_passed_entrance_exam(request, user, course):
+    """
+    Checks to see if the user has attained a sufficient score to pass the exam
+    """
+    if not course_has_entrance_exam(course):
+        return True
+    if user.is_anonymous():
+        return False
+    entrance_exam_score = get_entrance_exam_score(request, course)
+    if entrance_exam_score >= course.entrance_exam_minimum_score_pct:
+        return True
+    return False
+
+
+# pylint: disable=invalid-name
+def user_can_access_courseware_with_entrance_exam(request, user, course):
```

Comment:
I'm confused by the name of this function. The name indicates to me that it's assumed that `course` will have an entrance exam, yet we also double-check that assumption in the first line, returning a True value when the course does *not* have an entrance exam.